### PR TITLE
docs: add migration guide for docs deploy index action

### PR DESCRIPTION
**Pull request by Revathyvenugopal162**
continuation of #582

### CODE REVIEW

File: doc/source/migrations/index.rst [hunk 1]
```diff
@@ -25,6 +25,13 @@ Development version
   ``ansys/actions/doc-deploy-changelog``, ``ansys/actions/doc-deploy-dev``, and
   ``ansys/actions/doc-deploy-stable``.
 
+**Deprecated features:**
+
```
Comment (RobPasMue) [inline doc/source/migrations/index.rst:29]: We are missing references to action renaming (i.e. check-pr-title). Can somebody handle this?

**Review by RobPasMue — CHANGES_REQUESTED** [review]

File: doc/source/migrations/index.rst [hunk 2]
```diff
@@ -25,6 +25,13 @@ Development version
...
+- The ``ansys/actions/doc-deploy-index`` action has been deprecated and will be removed in the next release.
+  With the deprecation of ``pymeilisearch`` and the adoption of a static search index via the ``ansys-sphinx-theme``,
+  the ``ansys/actions/doc-deploy-index`` action is no longer necessary.
+
+
```
Comment (jorgepiloto) [inline doc/source/migrations/index.rst:34]: The ``ansys/actions/commit-style`` action has been renamed to ``ansys/actions/check-pr-title``.

File: doc/source/migrations/index.rst (hunk 1 above)
Comment (jorgepiloto) [reply doc/source/migrations/index.rst:29]: Just added this in a new item list, see https://github.com/ansys/actions/pull/586/commits/716c37e854fd70d687dd95dcd4868fe45c459839

File: doc/source/migrations/index.rst [hunk 3]
```diff
@@ -25,6 +25,13 @@ Development version
   ``ansys/actions/doc-deploy-changelog``, ``ansys/actions/doc-deploy-dev``, and
   ``ansys/actions/doc-deploy-stable``.
 
+**Deprecated features:**
+
+- The ``ansys/actions/doc-deploy-index`` action has been deprecated and will be removed in the next release.
+  With the deprecation of ``pymeilisearch`` and the adoption of a static search index via the ``ansys-sphinx-theme``,
+  the ``ansys/actions/doc-deploy-index`` action is no longer necessary.
+
+The ``ansys/actions/commit-style`` action has been renamed to ``ansys/actions/check-pr-title``.
```
Comment (Revathyvenugopal162) [inline doc/source/migrations/index.rst:34]:
```suggestion
- The ``ansys/actions/commit-style`` action has been renamed to ``ansys/actions/check-pr-title``.
```

**Review by RobPasMue — APPROVED** [review]
LGTM!

**Pull request merged**